src/nerve/frontends/tui/commander/monitor.py

```
@@ -0,0 +1,690 @@
+            with open(tmp_path, "w") as f:
+                f.write(full_content)
+
+            # Make the file read-only
+            os.chmod(tmp_path, 0o444)  # r--r--r--
+
+            # Get editor command with read-only flags
+            editor_cmd = self._get_editor_command()
+            editor_cmd.append(tmp_path)
+
+            # Open in editor (blocking call - wait for editor to close)
+            subprocess.run(editor_cmd, check=False)
```

Ruff (0.14.10)

`subprocess` call: check for execution of untrusted input

(S603)